scripts/scil_surface_create.py

```diff
@@ -0,0 +1,204 @@
+
+    assert_inputs_exist(parser, [], [args.in_labels, args.in_mask])
+    assert_outputs_exist(parser, args, args.out_surface)
+
+    assert args.out_surface.endswith('.vtk')
+
+    masks = []
+
+    if args.in_labels:
+        # Default value for isosurface
+        args.value = 0.5
+        # Load volume
+        img = nib.load(args.in_labels)
+        labels_volume = get_data_as_labels(img)
+
+        # Removed indices
+        if args.list_indices:
+            masks.append(merge_labels_into_mask(labels_volume,
+                                                ' '.join(args.list_indices)))
+        elif args.each_index:
+            indices = np.unique(labels_volume)[1:]
+            for index in indices:
+                masks.append(merge_labels_into_mask(labels_volume, str(index)))
+        else:
+            logging.warning('No indices provided, '
+                            'it will use all indices.')
+            masks.append(labels_volume > 0)
+    elif args.in_mask:
+        # Default value for isosurface
+        args.value = 0.5
+        # Load mask
+        img = nib.load(args.in_mask)
+        masks.append(get_data_as_mask(img))
+    else:
+        # Load volume
+        img = nib.load(args.in_volume)
+        masks.append(img.get_fdata(dtype=np.float32))
+
+    for it, mask in enumerate(masks):
+        # Basic morphology
+        if args.erosion is not None:
+            mask = binary_erosion(mask, iterations=args.erosion)
+        if args.dilation is not None:
+            mask = binary_dilation(mask, iterations=args.dilation)
+        if args.opening is not None:
+            mask = binary_opening(mask, iterations=args.opening)
+        if args.closing is not None:
+            mask = binary_closing(mask, iterations=args.closing)
+
+        if args.fill:
+            mask = binary_fill_holes(mask)
+
+        # Extract marching cube surface from mask
+        vertices, triangles = mcubes.marching_cubes(mask, args.value)
+
+        # Generate mesh
+        mesh = trimeshpy.trimesh_vtk.TriMesh_Vtk(triangles.astype(int),
+                                                 vertices)
+
+        # Transformation based on the Nifti affine
+        if not args.not_vox2vtk:
+            mesh.set_vertices(vtk_u.vox_to_vtk(mesh.get_vertices(),
+                                               img))
+
+        # Smooth
+        if args.smooth is not None:
+            new_vertices = mesh.laplacian_smooth(1, args.smooth,
+                                                 l2_dist_weighted=False,
+                                                 area_weighted=False,
+                                                 backward_step=True)
+            mesh.set_vertices(new_vertices)
+
+        if len(masks) == 1:
+            vtk_u.save_polydata(mesh.get_polydata(),
+                                args.out_surface,
+                                legacy_vtk_format=True)
+        else:
+            out_name = args.out_surface.replace('.vtk',
```

Comment:
Related to the other comment about format/extension, pick the extension of the out_surface and replace that instead of limiting it to vtk